security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -457,10 +462,6 @@ func recycleConnection(conID, resourceName string) {
 }
 
 func parseDiscoveryRequest(discReq *xdsapi.DiscoveryRequest) (string /*resourceName*/, error) {
-	if discReq.Node.Id == "" {
```

Comment:
Can we add a check of node.Id == "" here or before the call to parseDiscoveryRequest() and return error if Id is empty?

Reply:
Fixed to revert to the original structure.